QgisModelBaker/libqgsprojectgen/dbconnector/mssql_connector.py

```diff
@@ -0,0 +1,511 @@
+                disp_name_join = """
+                LEFT JOIN {}.t_ili2db_column_prop alias ON c.table_name = alias.tablename
+                    AND c.column_name = alias.columnname
+                    AND alias.tag = 'ch.ehi.ili2db.dispName'""".format(self.schema)
+                full_name_join = """
+                LEFT JOIN {}.t_ili2db_attrname full_name ON full_name.owner='{}'
+                    AND c.column_name=full_name.sqlname""".format(self.schema, table_name)
+            
+            # TODO Remove 'distinct' when issue 255 is solved
+            query = """
+                SELECT distinct c.column_name,
+                    case c.data_type when 'decimal' then 'numeric' else c.DATA_TYPE end as data_type,
+                    c.numeric_scale,
+                    {unit_field}
+                    {text_kind_field}
+                    {column_alias}
+                    {full_name_field}
+                    '------' AS comment
+                FROM INFORMATION_SCHEMA.COLUMNS AS c
+                {unit_join}
+                {text_kind_join}
+                {disp_name_join}
+                {full_name_join}
+                WHERE TABLE_NAME = '{table}'
+                    AND TABLE_SCHEMA = '{schema}'
+                """.format(schema=self.schema, table=table_name, unit_field=unit_field,
+                            text_kind_field=text_kind_field, column_alias=column_alias,
+                            full_name_field=full_name_field,unit_join=unit_join, text_kind_join=text_kind_join,
+                            disp_name_join=disp_name_join,
+                            full_name_join=full_name_join)
+            fields_cur.execute(query)
+            res = self._get_dict_result(fields_cur)
+            return res
+
+    def get_constraints_info(self, table_name):
+        result = {}
+        # Get all 'c'heck constraints for this table
+        if self.schema:
+            constraints_cur = self.conn.cursor()
+            
+            query = """
+                SELECT CHECK_CLAUSE
+                FROM
+                    INFORMATION_SCHEMA.CHECK_CONSTRAINTS cc INNER JOIN
+                    INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE c
+                        ON cc.CONSTRAINT_NAME = c.CONSTRAINT_NAME
+                        AND cc.CONSTRAINT_SCHEMA = c.CONSTRAINT_SCHEMA
+                WHERE
+                    cc.CONSTRAINT_SCHEMA = '{schema}'
+                    AND TABLE_NAME = '{table}'
+                """.format(schema=self.schema, table=table_name)
+            
+            constraints_cur.execute(query)
+
+            # Create a mapping in the form of
+            #
+            # fieldname: (min, max)
+            constraint_mapping = dict()
+            for constraint in constraints_cur:
+                m = re.match(r"\(\[(.*)\]>=\(([+-]?[0-9]+(?:\.[0-9]+)?(?:[eE][+-]?[0-9]+)?)\) AND \[(.*)\]<=\(([+-]?[0-9]+(?:\.[0-9]+)?(?:[eE][+-]?[0-9]+)?)\)\)", constraint[0])
```

Comment:
can you explain in words what does this regex and/or add a link to something like regex101 to show an example?

Reply:
This method (get_constraints_info) gets the intervals of some number fields. These intervals are stored as **check constraints** into the database.

The query in this method returns the **clause check** intervals that are similar to:

([**numero_pisos**]>=(**1**) AND [**numero_pisos**]<=(**100**))

So, the regex takes the query results and gets the field name, the minimum value, and the maximum value of the field for each register.
<img src='https://user-images.githubusercontent.com/29440924/63358943-7d07fc80-c331-11e9-9f29-c4e264653e68.png' width='70%'>

Reply:
perfect, I believe this would help to write this down in the code as a comment